crates/subspace-service/src/sync_from_dsn/snap_sync.rs

```diff
@@ -43,13 +60,15 @@ pub(crate) async fn snap_sync<Block, AS, Client, PG>(
     sync_service: Arc<SyncingService<Block>>,
     network_service_handle: NetworkServiceHandle,
     erasure_coding: ErasureCoding,
+    target_block_provider: Arc<dyn SnapSyncTargetBlockProvider>,
```

Comment:
What do you think about making this `Option<BlockNumber>` instead and decide how to orchestrate block number retrieval externally?

Specifically with other suggestions it'll be simply awaiting on a oneshot channel before starting snap sync, shouldn't need a while provider just for oneshot receiver.

Reply:
I don't think I follow you here. 

The actual `sync` function works with the `target_block: Option<BlockNumber` as you suggested, `snap_sync` wrapper handles the acquisition of the target block. We can move this acquisition up the call stack but I consider that more confusing. Also, the `target block provider` trait incapsulates the default non-blocking behavior of the consensus chain sync and IMO is better than the one-shot receiver because of it.